doc/migration_guide.rst

```diff
@@ -41,19 +48,118 @@ Functionality removed from the TLS implementation includes
 * DHE_PSK ciphersuites
 * CECPQ1 ciphersuites
 
-TLS 1.3 API adaptions
----------------------
+enum classes
+^^^^^^^^^^^^
+
+The publicly available C++ enums in the TLS namespace are now `enum class` and
+their member naming scheme was converted from `SHOUTING_SNAKE_CASE` to
+`CamelCase`.
+
+Callbacks
+^^^^^^^^^
+
+A number of new callbacks were added with TLS 1.3. None of those new callbacks
+is mandatory to implement by applications, though. Additionally there are a few
+backward incompatible changes in callbacks that might require attention by some
+applications:
+
+tls_verify_cert_chain()
+"""""""""""""""""""""""
+
+The parameter `ocsp_responses` is now a vector of `std::optional<OCSP::Response>`
+rather than potentially null `std::shared_ptr<OCSP::Response>`.
```

Comment:
This is confusingly worded IMO in that it sounds like we are guaranteeing that the `std::optional`s are not `nullopt`, which is not the case, right?

Might be easier to just strike the "rather than potentially null":

> The parameter `ocsp_responses`, which was previously `std::shared_ptr<OCSP::Response>`, is now 
`std::optional<OCSP::Response>`